models/forum.server.model.ts

```diff
@@ -1,169 +1,266 @@
  * @param params object containing forum post attributes
- * @param done function callback, returns status code, and message if error, or JSON if successful
  */
-export function insertPost(params, done) {
-  // Set forum post attributes
-  const userID = params.userID,
-    communityID = params.communityID,
-    title = params.title,
-    bodyText = params.bodyText,
-    edited = false,
-    upVotes = 0,
-    downVotes = 0,
-    attachments = params.attachments,
-    comments = [''];
-
+export async function insertPost(params: InsertPostDTO): Promise<PostDocument> {
   // Create new forum post document
-  const newPost = new Forum({
-    userID,
-    communityID,
-    title,
-    bodyText,
-    edited,
-    upVotes,
-    downVotes,
-    attachments,
-    comments,
+  const newPost = new Post({
+    owner: params.owner,
+    community: params.community,
+    title: params.title,
+    bodyText: params.bodyText,
+    edited: params.edited ?? false,
+    upVotes: params.upVotes ?? 0,
+    downVotes: params.downVotes ?? 0,
+    attachments: params.attachments ?? [],
+    comments: params.comments ?? [],
   });
 
   // Save new forum post document to database collection
-  newPost
-    .save()
-    .then((res) => {
-      return done(res);
-    })
-    .catch((err) => {
-      // Forum post is already in the database with unique attributes, return duplicate conflict error
-      if (err.code === 11000) {
-        return done({ err: 'Conflict', status: 409 });
-      }
-      // Any other database error, return internal server error
-      return done({ err: 'Internal server error', status: 500 });
-    });
+  try {
+    return await newPost.save();
+  } catch (err) {
+    // Forum post is already in the database with unique attributes, return duplicate conflict error
+    if (getProp(err, 'code') === 11000) {
+      throw new ServerError('Conflict', 409, err);
+    }
+    // Any other error
+    throw err;
+  }
+}
+
+export async function populatePost(post: PostDocument): Promise<PopulatedPost> {
+  return post.populate<{
+    owner: UserDocument;
+    community: CommunityDocument;
+  }>(['community', 'owner']);
+}
+
+export async function populatePosts(
+  posts: PostDocument[],
+): Promise<PopulatedPost[]> {
+  return (await Post.populate(posts, [
+    { path: 'community' },
+    { path: 'owner' },
+  ])) as PopulatedPost[];
 }
 
 /**
  * Search for a forum post in the database
  * @param id forum post ID
- * @param done function callback, returns status code, and message if error, or JSON if successful
  */
-export function searchPostById(id, done) {
-  try {
-    Forum.findById(id)
-      .then((res) => done(res))
-      .catch((err) => {
-        return done({ status: 404, err: err });
-      });
-  } catch (err) {
-    return done({ err: 'Internal server error', status: 500 });
+export async function searchPostById(
+  id: mongoose.Types.ObjectId,
+): Promise<PostDocument> {
+  const resource = await Post.findById(id);
+
```

Comment:
Similar suggestion see 
https://github.com/SE701-T5/Backend/pull/102/files#r840217139

Reply:
I have fixed these

Reply:
Sorry I misclicked, should be line 96 - 98